src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/primitives/PrimitiveNodeFactory.java

```diff
@@ -203,7 +201,7 @@ public static AbstractPrimitiveNode getOrCreateNamed(final CompiledCodeObject me
         if (nodeFactory != null) {
             return createNode(nodeFactory, location, numReceiverAndArguments);
         } else {
-            return null;
+            return new PrimExternalCallNode(moduleName, functionName, numReceiverAndArguments);
```

Comment:
Maybe we could do some checking here instead of always returning a node: it's probably a good idea to lookup the external call and cache the result. So maybe we could perform the check here and only return a `PrimExternalCallNode` iff the module and function are found.